src/d/actor/d_a_sea.cpp

```diff
@@ -266,49 +313,160 @@ void daSea_Init() {
 }
 
 /* 8015BAF8-8015BB60       .text daSea_ChkAreaBeforePos__Fff */
-void daSea_ChkAreaBeforePos(f32, f32) {
-    /* Nonmatching */
+bool daSea_ChkAreaBeforePos(f32 x, f32 z) {
+    if (l_cloth.mInitFlag == 0) {
+        return false;
+    }
+
+    if (l_cloth.mWaterHeightMgr.GetHeight(x, z) == 0 && l_cloth.mCullStopFlag != 0) {
+        return false;
+    }
+
+    return true;
 }
 
 /* 8015BB60-8015BBFC       .text daSea_ChkArea__Fff */
-bool daSea_ChkArea(f32, f32) {
-    /* Nonmatching */
+bool daSea_ChkArea(f32 x, f32 z) {
+    if (!daSea_ChkAreaBeforePos(x, z)) {
+        return false;
+    }
+
+    if (l_cloth.getMinX() < x && x < l_cloth.getMaxX() && l_cloth.getMinZ() < z && z < l_cloth.getMaxZ()) {
+        return true;
+    }
+
+    return false;
 }
 
+// Fakematch
+#pragma opt_dead_assignments off
 /* 8015BBFC-8015BDB0       .text daSea_calcWave__Fff */
-f32 daSea_calcWave(f32, f32) {
-    /* Nonmatching */
+f32 daSea_calcWave(f32 x, f32 z) {
+    if (!daSea_ChkArea(x, z)) {
+        return daSea_packet_c::BASE_HEIGHT;
+    }
+
+    f32 frac = 1.0f / 800.0f;
+
+    int x0 = (x - l_cloth.getMinX()) * frac;
+    int z0 = (z - l_cloth.getMinZ()) * frac;
+
+    f32* pY = l_cloth.mpHeightTable;
+    pY += x0;
+    pY += z0 * 65;
+
+    //f32 minX = (x0 * 800) + l_cloth.getMinX();
+    //f32 maxX = minX + 800.0f;
+    //f32 minZ = (z0 * 800) + l_cloth.getMinZ();
+    //f32 maxZ = minZ + 800.0f;
+    
+    Vec v00, v01, v10, v11;
+
+    v00.x = (x0 * 800) + l_cloth.getMinX();
+    v00.y = pY[0];
+    v00.z = (z0 * 800) + l_cloth.getMinZ();
+
+    v01.x = (x0 * 800) + l_cloth.getMinX();
+    v01.y = pY[65];
+    v01.z = v00.z + 800.0f;
+
+    v10.x = v01.x + 800.0f;
+    v10.y = pY[1];
+    v10.z = (z0 * 800) + l_cloth.getMinZ();
+
+    v11.x = v10.x;
+    v11.y = pY[66];
+    v11.z = v01.z;
+
+    Vec norm;
+    f32 baseY;
+
+    f32 f0, f1;
+    f1 = x - v01.x;
+    f0 = z - v10.z;
+    f1 *= frac;
+    f0 *= frac;
+
+    if (f1 + f0 >= 1.0f) {
+        cM3d_CalcPla(&v01, &v10, &v11, &norm, &baseY);
+    } else {
+        cM3d_CalcPla(&v00, &v01, &v10, &norm, &baseY);
+    }
+
+    return -((norm.x * x) + (norm.z * z) + baseY) / norm.y;
 }
 
+#pragma opt_dead_assignments reset
+
 /* 8015BDB0-8015C010       .text daSea_GetPoly__FPvPFPvR4cXyzR4cXyzR4cXyz_vRC4cXyzRC4cXyz */
-void daSea_GetPoly(void*, void (*)(void*, cXyz&, cXyz&, cXyz&), const cXyz&, const cXyz&) {
-    /* Nonmatching */
+void daSea_GetPoly(void* param_1, void (*callback)(void*, cXyz&, cXyz&, cXyz&), const cXyz& param_3, const cXyz& param_4) {
+    if (!daSea_ChkArea(param_3.x, param_3.z) || !daSea_ChkArea(param_4.x, param_4.z)) return;
+
+    f32 frac = 1.0f / 800.0f;
+    int x0 = (param_3.x - l_cloth.getMinX()) * frac;
+    int z0 = (param_3.z - l_cloth.getMinZ()) * frac;
+    int x1 = (param_4.x - l_cloth.getMinX()) * frac;
+    int z1 = (param_4.z - l_cloth.getMinZ()) * frac;
```

Comment:
My names here would be

param_1 = pUserData (or userArea/userWork, which tend to be more common in WW's codebase)
param_3 = minPt
param_4 = maxPt